### ISSUE TITLE
Prevent Device Sleep

### ISSUE THREAD

**Issue by tneil** [blackberry/WebWorks#25]
Prevent device going to sleep (some app like chess clock need the screen always on)

**Comment by tneil** [comment]
There's an API in the WebWorks-Community-APIs that could serve as a starting point

https://github.com/blackberry/WebWorks-Community-APIs/tree/master/Smartphone/Sleep

**Comment by rtholmes** [comment]
It would be great if there was a real WebWorks API for this. I am wary of using the extension because it means my app interferes with the user's backlight timer value if they have set it to anything other than the default.

**Comment by tneil** [comment]
This was the only AIR API that we could find that would prevent the sleep. There was an other API that was available but it wasn't functioning properly on the version of PlayBook we were testing on

**Comment by rtholmes** [comment]
That's too bad. It looks like SystemIdleMode.KEEP_AWAKE could work, but I'm guessing that's the one you tried first but didn't work.

**Comment by tneil** [comment]
Yes.. I believe Adam Stanley filed a bug with the AIR team that it wasn't working properly

**Comment by rtholmes** [comment]
Great, thanks Tim; I have had users requesting this feature so I'm glad to hear it's being looked at in some way!

**Comment by raakhtar** [comment]
Hello Neil,

Any update on KEEP_AWAKE with Playbook OS 2.0. Would you be able to point me in the right direction. A lot of users of my PlayBook & Blackberry Smartphone Webworks App is asking me for this functionality and they compare it with Native alarm apps and its possibility. Its difficult to provide them excuse. Therefore, looking for you help.

**Comment by tneil** [comment]
Not sure.. I'll ask and see if anyone has an update

**Comment by raakhtar** [comment]
Thanks for responding... Please also point me in the direction to obtain similar behavior on Blackberry Smartphone on Webworks application. It's very imperative to have alarm notification from Webworks application runing on Blackberry Smartphone at the moment

**Comment by tneil** [comment]
Are you looking for how to keep your app running in the background on BB Smartphones?  Or are you looking at making the screen not dim after a period of inactivity?

**Comment by rtholmes** [comment]
I have three apps that need to not dim. This functionality is available in:
- iOS ([UIApplication sharedApplication].idleTimerDisabled = YES; )
- Android (getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);)
- WebOS (blockScreenTimeout : true)
- WP7 (PhoneApplicationService.Current.UserIdleDetectionMode = IdleDetectionMode.Disabled;)

And is the main shortcoming of webworks compared to other platforms (for the apps I build). (There are many strengths of the  platform as well, this is just my main stumbling block)

**Comment by tneil** [comment]
I believe there's a community API that will prevent sleep on the smartphone

https://github.com/blackberry/WebWorks-Community-APIs/tree/master/Smartphone/Sleep

**Comment by raakhtar** [comment]
Hello tim,

This is not what I am looking for this would simpy force BB backlight to stay lit. I have an appworld application which needs to play an audio file after a certain interval(prayer timings for the day). If the device is active and in use. The alarm fires up, however, if device is in sleep mode the alarm does not fire up... Any help would be greatly appreciated as user's are having bad experience eventually effecting rating of my app...looking for ur help

Thanks
Regards
Raheal

----- Original Message -----

**Comment by tneil** [comment]
Just to be clear.. are you talking about BB smartphones? or PlayBook? Or you are having problems with this on both?

This should definitely be attainable on BB smartphones with WebWorks.  You simply send your application to the background to run.

Use requestBackground() from the application object https://bdsc.webapps.blackberry.com/html5/apis/blackberry.app.html

**Comment by raakhtar** [comment]
I am certainly talking about both smartphone and playbook however smartphone has a bigger user base that's why getting the app running perfectly fine on smartphone is top most priority at the moment due to competitor apps in native Java and having this functionaly rightly placed.

Could you kindly point me to any references inorder for me to see how this gets accomplished for smartphones to start with atleast.

Thanks for your help
Regards
Raheal

----- Original Message -----

**Comment by tneil** [comment]
For smartphones...

To have your application run on start-up you can check out this content:
https://bdsc.webapps.blackberry.com/html5/documentation/ww_developing/rim_background_element_1265316_11.html

And for sending your application to the background, the link in my last point will accomplish that task.

**Comment by raakhtar** [comment]
Much appreciated Tim for the links and prompt support. Kudos +1

Regards
Raheal

----- Original Message -----

**Comment by raakhtar** [comment]
Tim would you also comment on its availabilty starting from which version of webworks sdk for smartphone since I am using 2.5 version of eclipse plugin for webworks as the app crashes severly using 2.3 sdk. Therefore I had reverted to the eclipse plugin version of webworks sdk, don't know which version is this. Would you be able to also shed some light on it.

Regards
Raheal

----- Original Message -----

**Comment by tneil** [comment]
I'm trying to remember when it was available...  I'm not sure if it was available in the versions supported by the Eclipse plug-in or not.  

I would always recommend getting the latest SDK.. We do all our optimizing, bug fixes and memory management improvements in the latest code.
